libs/hdf-converters/src/conveyor-mapper.ts

```diff
@@ -0,0 +1,244 @@
+const DEFAULT_NIST_TAG = ['SI-2', 'RA-5'];
+
+function createDescription(
+  data: Record<string, unknown>,
+  score: number,
+  date: string,
+  type: string
+): Record<string, unknown> {
+  let desc = '';
+  if (type == 'Moldy') {
+    desc =
+      'body:' +
+      _.get(data, 'body') +
+      '\nbody_format:' +
+      (_.get(data, 'body_format') as string) +
+      '\nclassificaton:' +
+      (_.get(data, 'classification') as string) +
+      '\n';
+  } else if (type == 'Stigma') {
+    desc =
+      'body:' +
+      _.get(data, 'body') +
+      '\nbody_format:' +
+      (_.get(data, 'body_format') as string) +
+      '\ntitle_text:' +
+      (_.get(data, 'title_text') as string) +
+      '\nclassificaton:' +
+      (_.get(data, 'classification') as string) +
+      '\n';
+  } else if (type == 'CodeQuality') {
+    desc =
+      'body:' +
+      _.get(data, 'body') +
+      '\nbody_format:' +
+      (_.get(data, 'body_format') as string) +
+      '\nclassificaton:' +
+      (_.get(data, 'classification') as string) +
+      '\ndepth:' +
+      (_.get(data, 'depth') as string) +
+      '\nheuristic:' +
+      (_.get(data, 'heuristic') as string) +
+      '\ntitle_text:' +
+      (_.get(data, 'title_text') as string) +
+      '\n';
+  } else {
+    desc = JSON.stringify(data);
+  }
+  return {
+    status: determineStatus(score),
+    code_desc: desc,
+    start_time: date
+  };
+}
+
+function childrenfinder(output: Record<string, unknown>): string[][] {
+  const arr: string[][] = [];
+  _.forEach(output, (value, key) => {
+    if (_.has(value, 'name')) {
+      const temp: string = _.get(value, 'name[0]') || '';
+      arr.push([key, temp]);
+    }
+    if (_.has(value, 'children')) {
+      const temp: Record<string, unknown> = _.get(value, 'children') || {};
+      _.forEach(childrenfinder(temp), function (value) {
+        arr.push(value);
+      });
+    }
+  });
+  return arr;
+}
+function shafileMapper(
+  output: Record<string, unknown>
+): Record<string, unknown> {
+  const toplevel = _.get(output, 'api_response.file_tree') as Record<
+    string,
+    unknown
+  >;
+  const shamappings = {};
+  _.forEach(toplevel, (value) => {
+    if (_.has(value, 'name')) {
+      const temp: string = _.get(value, 'name[0]') || '';
+      _.set(shamappings, temp, _.get(value, 'sha256'));
+    }
+    if (_.has(value, 'children')) {
+      const temp: Record<string, unknown> = _.get(value, 'children') || {};
+      const arr = childrenfinder(temp);
+      _.forEach(arr, (value) => {
+        const temp1: string = value[0] || '';
+        const temp2: string = value[1] || '';
+        _.set(shamappings, temp1, temp2);
+      });
+    }
+  });
+  return shamappings;
+}
+function determineStatus(score: number): ExecJSON.ControlResultStatus {
+  if (score == 0) {
+    return ExecJSON.ControlResultStatus.Passed;
+  }
+  return ExecJSON.ControlResultStatus.Failed;
+}
+
+function arrayifyObject(
```

Comment:
ideally we minimize the amount of pre-processing we need to do since it makes it harder for maintainers to come back later on and understand how the mapping works since the paths/data that the mapper uses are not actually analogous to the original data.

you said that some of this was necessary because the controls block was expecting 'code_desc' and other attributes at that path which is strange to me since that shouldn't be the case.  when you're working on resolving this issue, reach out to me and we can do a working session.

Reply:
for future reference, we tried implementing this and realized we needed access to the parent attribute in order to actually make this work.  currently the baseconverter doesn't support this behavior (however reversebaseconverter does which is ever more reason to combine those classes) so we are forced to do the implementation shown here.